docs/manual/src/udl/ext_types_wrapped.md

```diff
@@ -57,13 +83,86 @@ Because `UniffiCustomTypeWrapper` is defined in each crate, this means you can w
 if they are not defined in your crate - see the 'wrapper_types' example which demonstrates
 how `serde_json::Value` can be wrapped.
 
-## Custom Foreign Wrappers
+## Bindings example
+
+Suppose you have Rust struct with recently renamed fields:
+
+```rust
+pub struct User {
+  full_name: String,
+  age: Int32,
+}
+```
+
+But your Kotlin code is still using the old field names:
+
+```kotlin
+data class LegacyUser(
+  var name: String,
+  var years_old: Int,
+}
+```
+
+We again start by declaring this type in our UDL via a `typedef` with a `Wrapped` attribute.
+
+```idl
+[Wrapped]
+typedef User LegacyUser;
+```
+
+This time however, we will update the Kotlin section of the `uniffi.toml` file:
+
+```toml
+[bindings.kotlin.wrappers.LegacyUser]
+# Name of the Kotlin class that will wrap `User`
+type_name = "LegacyUser"
+# Function that converts `User` to `LegacyUser`
+wrap = "wrapUser({})"
+# Function that converts `LegacyUser` to `User`
+unwrap = "unwrapUser({})"
+# Where we can import those functions from
+imports = [
+  "my.wrappers.package.wrapUser",
+  "my.wrappers.package.unwrapUser",
+]
+```
+
+`wrap` and `unwrap` specifies the code used to wrap/unwrap the type.  `{}`
+will be replaced with the value to be wrapped or unwrapped.
+
+`my.wrappers.package` would have code like this:
+
+```kotlin
+fun wrapUser(user: User) = LegacyUser(
+   name = user.fullName,
+   yearsOld = user.age,
+)
+
+fun unwrapUser(user: LegacyUser) = User(
+   fullName = user.name,
+   age = user.yearsOld,
+)
+```
+
+## Wrapping on both the scaffolding and bindings side
+
+Of course, it's possible to wrap on both the scaffolding and bindings side.
```

Comment:
nit: drop "Of course" - it's like saying "Obviously", which often isn't :) Also, this just seems to restate what's at the start of this section - eg, the sentence "allows you to define types that feel like new builtin types" is important, but it's buried here. This para and the discussion about the json should either be deleted or moved up where you talk about the handle. (before your patch, the handle and json object discussions were very close - now the whole "user" stuff is between them, which seems wrong)